README.md

```diff
@@ -1 +1,404 @@
+
+## Break (10 mins)
+
+## GET data from an external API using AJAX
+
+### I Do: GET example (10 mins)
+
+Let's build a very simple app that posts a movie title and poster after searching for it!
+[OMDB movie search](https://github.com/joe-gz/omdb-api)
+
+The starter code is linked above. It contains a basic HTML/CSS/JS setup. If you open up the HTML in the browser, you will notice that searching for something returns no results.
+
+Let's go ahead and add in the AJAX request:
+
+```js
+// get value from search input field
+var keyword = $("input[name='keyword']").val();
+var url = "https://www.omdbapi.com/?t="+keyword
+$.ajax({
+  url: url,
+  type: "GET",
+  dataType: "json"
+}).done ( function(response){
+  console.log(response);
+  // call movie function below to append movie titles
+  movie(response);
+}).fail ( function (){
+  console.log("fail");
+}).always( function(){
+  console.log("Something happens");
+})
+```
+
+**What does all of this mean?**
+
+$.ajax takes an object as an argument with at least three key-value pairs...
+(1) The URL endpoint for the JSON object.
+(2) Type of HTTP request.
+(3) Datatype. Usually JSON.
+
+The `url` that we've included is the api url from OMDB, with the added keyword that we are searching for. The `type` is simply the verb we want performed; in this case GET. And, the `datatype` will typically be `json`.
+
+### Promises
+
+Following `datatype`, there are three methods that we call promises.  In short, these tell the `$.ajax` request what to do if the request is successful, fails, or both.
+
+* `.done` requires a callback that determines what we do after a successful AJAX call
+* `.fail` requires a callback that determines what we do after an unsuccessful AJAX call
+* `.always` requires a callback that determines what we do regardless of a successful or unsuccessful call
+
+**Bonus**: you'll notice that if you simply search for "Star Wars", it doesn't really work! why is that? Is there another url you can use from OMDB API that would return all Star Wars movies?
+
+### You Do: GET from Tunr (25 mins)
+
+Fork and clone another [Tunr](https://github.com/ga-wdi-exercises/tunr_rails_ajax) repo!
+Once you've cloned the repo, `cd` into it and run the usual commands...
+
+```bash
+$ bundle install
+$ rake db:drop
+$ rake db:create db:migrate db:seed
+```
+
+We can now use `$.ajax()` to make asynchronous HTTP requests to our Tunr app! Let's go ahead and create a new Artists controller action and corresponding view: `test_ajax`
+
+#### Setting up a view to test AJAX with
+Let's update our routes in `config/routes.rb` for a new route to test all of our AJAX calls in:
+
+```ruby
+get '/test_ajax', to: 'artists#test_ajax'
+```
+
+in `app/controllers/artists_controller.rb`:
+
+```ruby
+# We're just creating this so we can test AJAX on a separate view, test_ajax.html.erb.
+def test_ajax
+end
+```
```

Comment:
Could be a good opportunity to draw out a tiny portion of the MVC diagram (or perhaps Matt's more-detailed version of it). Not sure how much experience the students have had with creating custom routes / controller actions.